relayer/pkg/starknet/client.go

```diff
@@ -65,21 +70,14 @@ func NewClient(chainID string, baseURL string, lggr logger.Logger, timeout *time
 		client.defaultTimeout = *timeout
 	}
 
-	client.set(baseURL, chainID) // hack: change the base URL & chainID (not supported in caigo)
-
-	return client, nil
-}
-
-func (c *Client) set(baseURL, chainID string) {
-	if chainID != "" {
-		c.Gw.Gateway.ChainId = chainID // note: gateway API in caigo does not query endpoint, uses what is set
+	// cache chainID on the provider to avoid repeated calls
+	chainID, err := client.Provider.ChainID(context.TODO())
```

Comment:
Note: I did this to avoid `ChainID` calls, but I noticed that any time `NewAccount` is called, this happens internally in caigo too. Since the TXM creates accounts all the time, this would be called on any submit:

https://github.com/smartcontractkit/caigo/blob/56b86bbbc0c4a78bba811a5c25088c5d4184c031/account.go#L110-L114

We should patch this in caigo and cache on the provider instead